Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM ubuntu:trusty
+MAINTAINER Di Wu <diwu@yelp.com>
+
+COPY . /opt/redash/current/
+
+WORKDIR /opt/redash/current
+
+ENV FILES_BASE_URL /opt/redash/current/setup/files/
+
+# Base packages
+RUN apt-get update && \
+  apt-get install -y python-pip python-dev curl build-essential pwgen libffi-dev sudo git-core
+RUN pip install -U setuptools
+
+# redash user
+RUN useradd --system --comment " " --create-home redash
+RUN useradd --system --comment " " --create-home postgres
+
+# Install dependencies via apt-get
+RUN apt-get update && \
+  apt-get -y install libpq-dev postgresql-client
+
+# Make logs folder
+RUN mkdir /opt/redash/logs
+
+# Default config file
+RUN cp $FILES_BASE_URL"env" /opt/redash/.env
+
+# Install dependencies
+RUN cd /opt/redash/current
+RUN pip install -r requirements.txt
+
+RUN apt-get install -y libssl-dev && \
+  apt-get install -y libmysqlclient-dev
+
+# Pip requirements for all data source types
+RUN pip install -r requirements_all_ds.txt
+
+# Setup supervisord + sysv init startup script
+RUN mkdir -p /opt/redash/supervisord
+RUN pip install supervisor==3.1.2
+
+# Get supervisord startup script
+RUN cp $FILES_BASE_URL"supervisord_docker.conf" /opt/redash/supervisord/supervisord.conf
+
+RUN cp $FILES_BASE_URL"redash_supervisord_init" /etc/init.d/redash_supervisord
+
+# Fix permissions
+RUN chown -R redash /opt/redash
+
+CMD bash /etc/init.d/redash_supervisord start
```

Comment:
The tar ball isn't just the packaged source code. It has the front end
assets already "compiled", which saves the need to bloat the image with
Node.js dependencies that are not needed in production.

So what I had in mind is to have a base image which we use for production.
And another image for development that extends it with the Node.js stack
and uses redash sources from an attached volume that will be mapped to
user's local source code (the use of volume instead of copy is to avoid the
need to rebuild the image everytime you edit the source code).

But if you prefer to start simpler with one image, without volumes, that
uses local files to build the image that's fine too. Because I can probably
run the Docker image creation from the CI machine. Just replace the tar
ball download with the copy command. The rest of my suggestions still make
sense, no?
On Oct 7, 2015 9:25 PM, "John Wu" <notifications@github.com> wrote:

> In Dockerfile
> <https://github.com/EverythingMe/redash/pull/588#discussion_r41426642>:
>
> > +# Pip requirements for all data source types
> > +RUN pip install -r requirements_all_ds.txt
> > +
> > +# Setup supervisord + sysv init startup script
> > +RUN mkdir -p /opt/redash/supervisord
> > +RUN pip install supervisor==3.1.2
> > +
> > +# Get supervisord startup script
> > +RUN cp $FILES_BASE_URL"supervisord_docker.conf" /opt/redash/supervisord/supervisord.conf
> > +
> > +RUN cp $FILES_BASE_URL"redash_supervisord_init" /etc/init.d/redash_supervisord
> > +
> > +# Fix permissions
> > +RUN chown -R redash /opt/redash
> > +
> > +CMD bash /etc/init.d/redash_supervisord start
>
> Not exactly. I think there is some miscommunication. Because I don't see
> any difference between copy from local source and download from latest
> code, besides that COPY allows user to build their own images based on
> their code, which doesn't change our workflow at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/EverythingMe/redash/pull/588/files#r41426642>.
>